R/unit.R

```diff
@@ -816,48 +628,58 @@ grobDescent.default <- function(x) {
 # on parent's drawing context or size)
 #########################
 
-# Only deals with unit of length() 1
-absolute <- function(unit) {
-  !is.na(match(attr(unit, "unit"),
-               c("cm", "inches", "lines", "null",
-                 "mm", "points", "picas", "bigpts",
-                 "dida", "cicero", "scaledpts",
-                 "strwidth", "strheight", "strascent", "strdescent", "char",
-                 "mylines", "mychar", "mystrwidth", "mystrheight",
-                 # pseudonyms (from unit.c)
-                 "centimetre", "centimetres", "centimeter",  "centimeters",
-                 "in", "inch",
-                 "line",
-                 "millimetre", "millimetres", "millimeter", "millimeters",
-                 "point", "pt")))
-}
-
-# OLD absolute.unit
 absolute.units <- function(unit) {
-  UseMethod("absolute.units")
-}
-
-absolute.units.unit <- function(unit) {
-  n <- length(unit)
-  new.unit <- if (absolute(unit[1L])) unit[1L] else unit(1, "null")
-  if(n > 1) for(i in 2L:n)
-    new.unit <- unit.c(new.unit, absolute.units(unit[i]))
-  new.unit
-}
-
-absolute.units.unit.list <- function(unit) {
-  structure(class = class(unit),
-            lapply(unit, absolute.units))
-}
-
-absolute.units.unit.arithmetic <- function(unit) {
-  switch(unit$fname,
-         "+"=unit.arithmetic("+", absolute.units(unit$arg1),
-           absolute.units(unit$arg2)),
-         "-"=unit.arithmetic("-", absolute.units(unit$arg1),
-           absolute.units(unit$arg2)),
-         "*"=unit.arithmetic("*", unit$arg1, absolute.units(unit$arg2)),
-         "min"=unit.arithmetic("min", absolute.units(unit$arg1)),
-         "max"=unit.arithmetic("max", absolute.units(unit$arg1)),
-         "sum"=unit.arithmetic("sum", absolute.units(unit$arg1)))
-}
+  .Call(C_absoluteUnits, unit)
+}
+
+# Lookup table for unit ids
```

Comment:
If we must have this, make sure there is a comment that it MUST match the C enumeration in grid.h

Reply:
Will do